contracts/libraries/ShareMath.sol

```diff
@@ -71,7 +72,7 @@ library ShareMath {
 
             unredeemedShares = uint128(unredeemedShares256);
         } else {
-            unredeemedShares = depositReceipt.unredeemedShares;
+            unredeemedShares = defaultSharePrice.mul(depositReceipt.unredeemedShares);
```

Comment:
Needs to rename to `initialSharePrice`

Reply:
Think we have to retain the old statement:

```solidity
unredeemedShares = depositReceipt.unredeemedShares;
```

Instead, we should add another condition that checks that it's the first round:

```solidity
} else if (depositReceipt.round == 1) {
  unredeemedShares = initiatalSharePrice.mul(depositReceipt.unredeemedShares);
} else {
  unredeemedShares = depositReceipt.unredeemedShares;
}
```

Reply:
I original misunderstood the logic in getSharePrice, and have moved the initialSharePrice adjustment to the rollover logic in `vaultLifecycle.sol`, where it's used when currentSupply is 0.